mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -3321,6 +3366,91 @@ void IfOp::print(OpAsmPrinter &p) {
   p.printOptionalAttrDict((*this)->getAttrs());
 }
 
+LogicalResult IfOp::verify() {
+  if (getThenGraph().empty() || getElseGraph().empty())
+    return emitOpError("require `then_graph` and `else_graph` not be empty");
+
+  if (errorIfTensorListShapeMismatch(
+          *this, getThenGraph().front().getArguments(),
+          "`then_graph` arguments", getInputList(), "`input_list`")
+          .failed())
+    return failure();
+
+  if (errorIfTensorListShapeMismatch(
+          *this, getElseGraph().front().getArguments(),
+          "`else_graph` arguments", getInputList(), "`input_list`")
+          .failed())
+    return failure();
+
+  auto thenYield = cast<tosa::YieldOp>(getThenGraph().front().getTerminator());
```

Comment:
Is it possible for the terminator to not be a "tosa.yield" op?

Reply:
I'm afraid not. TOSA WHILE and IF must have a TOSA YIELD as a terminator of control-flow block structure. Similar to RETURN as a terminator of function. For more detail, there is a code and test to ensure this property in mlir/lib/IR/Verifier.cpp and mlir/test/IR/invalid.mlir